EncryptedDatabaseErrorTest Failing

Tracking bug for the currently failing CI.

e.g.
https://github.com/google/android-fhir/runs/5077468637?check_suite_focus=true

```
com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest > unencryptedDatabase_thenEncryptedDatabase_shouldThrowIllegalStateException[test(AVD) - 10] FAILED 
	java.lang.AssertionError: unexpected exception type thrown; expected:<java.lang.IllegalStateException> but was:<java.lang.OutOfMemoryError>
	at org.junit.Assert.assertThrows(Assert.java:1020)

com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest > encryptedDatabase_thenUnencrypted_shouldThrowIllegalStateException[test(AVD) - 10] FAILED 
	java.lang.AssertionError: unexpected exception type thrown; expected:<java.lang.IllegalStateException> but was:<java.lang.OutOfMemoryError>
	at org.junit.Assert.assertThrows(Assert.java:1020)

com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest > encryptedDatabase_thenLostKey_shouldThrowSQLiteException[test(AVD) - 10] FAILED 
	java.lang.AssertionError: unexpected exception type thrown; expected:<net.sqlcipher.database.SQLiteException> but was:<java.lang.OutOfMemoryError>
	at org.junit.Assert.assertThrows(Assert.java:1020)

com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest > encryptedDatabase_thenLostKey_recreationStrategy_shouldRecreateDatabase[test(AVD) - 10] FAILED 
java.lang.OutOfMemoryError: OutOfMemoryError thrown while trying to throw OutOfMemoryError; no stack trace available
```

We use emulator to run these test and in the past we ran into issues when there's a new emulator release: https://github.com/ReactiveCircus/android-emulator-runner/issues/160

Interestingly, when I ran these tests locally on Nokia 4.2 (directly in android studio, or using ./gradlew connectedCheck), I can get some test failures (not exactly the same as the CI failures):

[getOrCreatePassphrase_otherAliasExists_shouldReplaceKey](https://github.com/google/android-fhir/blob/c06c2a3c29fedd25865e63d310ffaf88befebd62/engine/src/androidTest/java/com/google/android/fhir/db/impl/DatabaseEncryptionKeyProviderTest.kt#L57)

```
--------- beginning of main
02-07 11:13:26.014 +0000  1017  1002  1002 I keystore: del USRPKEY_test_key 10197
02-07 11:13:26.015 +0000  1017  1002  1002 I keystore: del USRSKEY_test_key 10197
02-07 11:13:26.016 +0000  1017  1002  1002 I keystore: del USRCERT_test_key 10197
02-07 11:13:26.017 +0000  1017  1002  1002 I keystore: del CACERT_test_key 10197
02-07 11:13:26.017 +0000  1017  1002  1002 I keystore: del USRPKEY_other_test_key 10197
02-07 11:13:26.018 +0000  1017  1002  1002 I keystore: del USRSKEY_other_test_key 10197
02-07 11:13:26.019 +0000  1017  1002  1002 I keystore: del USRCERT_other_test_key 10197
02-07 11:13:26.020 +0000  1017  1002  1002 I keystore: del CACERT_other_test_key 10197
--------- beginning of system
02-07 11:13:28.924 +0000  1000  1549  1549 D CompatibilityChangeReporter: Compat change id reported: 135634846; UID 10087; state: DISABLED
02-07 11:13:28.924 +0000  1000  1549  1747 D CompatibilityChangeReporter: Compat change id reported: 143937733; UID 10087; state: ENABLED
02-07 11:13:28.929 +0000  1000  1549  1549 D CompatibilityInfo: mCompatibilityFlags - 0
02-07 11:13:28.929 +0000  1000  1549  1549 D CompatibilityInfo: applicationDensity - 300
02-07 11:13:28.929 +0000  1000  1549  1549 D CompatibilityInfo: applicationScale - 1.0
02-07 11:13:28.934 +0000 10010 21411 21411 I Finsky  : [2] qdl.onStartJob(3): SCH: job service start with id 9521.
02-07 11:13:28.945 +0000  root   621   621 D Zygote  : Forked child process 23205
02-07 11:13:28.950 +0000  1000  1549  1747 I ActivityManager: Start proc 23205:com.hmdglobal.support/u0a87 for service {com.hmdglobal.support/androidx.work.impl.background.systemjob.SystemJobService}
02-07 11:13:28.963 +0000 10010 21411 23114 I Finsky  : [1578] qbu.a(90): SCH: Satisfied jobs for 9521 are: 12-1
02-07 11:13:28.966 +0000 10010 21411 23211 I Finsky  : [1585] psh.accept(16): SCH: Job 12-1 starting
02-07 11:13:28.968 +0000 10010 21411 21411 I Finsky  : [2] ContentSyncJob.v(1): [ContentSync] job started
02-07 11:13:28.975 +0000 10087 23205 23205 E dglobal.suppor: Not starting debugger since process cannot load the jdwp agent.
02-07 11:13:28.988 +0000  1000  1549  5841 D CompatibilityInfo: mCompatibilityFlags - 0
02-07 11:13:28.988 +0000  1000  1549  5841 D CompatibilityInfo: applicationDensity - 300
02-07 11:13:28.988 +0000  1000  1549  5841 D CompatibilityInfo: applicationScale - 1.0
02-07 11:13:28.990 +0000  1000  1549  5841 D CompatibilityInfo: mCompatibilityFlags - 0
02-07 11:13:28.990 +0000  1000  1549  5841 D CompatibilityInfo: applicationDensity - 300
02-07 11:13:28.990 +0000  1000  1549  5841 D CompatibilityInfo: applicationScale - 1.0
02-07 11:13:29.007 +0000 10087 23205 23205 I dglobal.suppor: The ClassLoaderContext is a special shared library.
02-07 11:13:29.040 +0000 10010 21411 23213 I Finsky  : [1587] tw.s(1): Completed 0 account content syncs with 0 successful.
02-07 11:13:29.040 +0000 10010 21411 21411 I Finsky  : [2] ContentSyncJob.a(1): [ContentSync] Installation state replication succeeded.
02-07 11:13:29.041 +0000 10010 21411 21411 I Finsky  : [2] qbs.r(4): SCH: jobFinished: 12-1. TimeElapsed: 74ms.
02-07 11:13:29.096 +0000 10010 21411 21454 I Finsky  : [1465] pdi.accept(30): SCH: Scheduling phonesky job Id: 1-1337, CT: 1644219769297, Constraints: [{ L: 17450037, D: 60650037, C: 1, I: 1, N: 1 }]
02-07 11:13:29.099 +0000 10087 23205 23205 W dglobal.suppor: JIT profile information will not be recorded: profile file does not exist.
02-07 11:13:29.100 +0000 10087 23205 23205 W dglobal.suppor: JIT profile information will not be recorded: profile file does not exist.
02-07 11:13:29.111 +0000 10087 23205 23205 I Perf    : Connecting to perf service.
02-07 11:13:29.128 +0000 10010 21411 21440 I Finsky  : [1453] pvj.apply(14): SCH: Scheduling 1 system job(s)
02-07 11:13:29.129 +0000 10010 21411 21440 I Finsky  : [1453] pvj.apply(20): SCH: Scheduling system job Id: 9524, L: 4810205, D: 48010205, C: false, I: false, N: 1
02-07 11:13:29.131 +0000 10087 23205 23205 D NetworkSecurityConfig: No Network Security Config specified, using platform default
02-07 11:13:29.131 +0000 10087 23205 23205 D NetworkSecurityConfig: No Network Security Config specified, using platform default
02-07 11:13:29.147 +0000 10010 21411 23211 I Finsky  : [1585] qdl.a(16): SCH: job service finished with id 9521.
02-07 11:13:29.241 +0000 10087 23205 23205 I FirebaseApp: Device unlocked: initializing all Firebase APIs for app [DEFAULT]
02-07 11:13:29.246 +0000 10087 23205 23231 V FA      : Deferring to Google Analytics for Firebase for event data collection. https://goo.gl/J1sWQy
02-07 11:13:29.256 +0000 10035  3482  3482 D BoundBrokerSvc: onUnbind: Intent { act=com.google.android.gms.feedback.internal.IFeedbackService cmp=com.google.android.gms/.chimera.GmsBoundBrokerService }
02-07 11:13:29.285 +0000 10087 23205 23205 I FirebaseCrashlytics: Initializing Firebase Crashlytics 18.2.1 for com.hmdglobal.support
02-07 11:13:29.286 +0000  logd     0     0 W swapper/5: type=1400 audit(0.0:5111): avc: denied { kill } for capability=5 scontext=u:r:kernel:s0 tcontext=u:r:kernel:s0 tclass=capability permissive=0
02-07 11:13:29.296 +0000 10087 23205 23231 I DynamiteModule: Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
02-07 11:13:29.296 +0000 10087 23205 23231 I DynamiteModule: Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
02-07 11:13:29.296 +0000 10087 23205 23231 V DynamiteModule: Dynamite loader version >= 2, using loadModule2NoCrashUtils
02-07 11:13:29.300 +0000 10087 23205 23234 I DynamiteModule: Considering local module com.google.android.gms.measurement.dynamite:51 and remote module com.google.android.gms.measurement.dynamite:59
02-07 11:13:29.300 +0000 10087 23205 23234 I DynamiteModule: Selected remote version of com.google.android.gms.measurement.dynamite, version >= 59
02-07 11:13:29.300 +0000 10087 23205 23234 V DynamiteModule: Dynamite loader version >= 2, using loadModule2NoCrashUtils
02-07 11:13:29.328 +0000 10087 23205 23205 W Glide   : Failed to find GeneratedAppGlideModule. You should include an annotationProcessor compile dependency on com.github.bumptech.glide:compiler in your application and a @GlideModule annotated AppGlideModule implementation or LibraryGlideModules will be silently ignored
02-07 11:13:29.349 +0000 10087 23205 23205 I TetheringManager: registerTetheringEventCallback:com.hmdglobal.support
02-07 11:13:29.351 +0000 10087 23205 23205 I FirebaseInitProvider: FirebaseApp initialization successful
02-07 11:13:29.361 +0000 10087 23205 23205 D WM-WrkMgrInitializer: Initializing WorkManager with default configuration.
02-07 11:13:29.371 +0000 10087 23205 23205 I AppStateListener:  OnCreate FreshchatInitProvider
02-07 11:13:29.382 +0000 10087 23205 23205 W com.facebook.q: The value for AdvertiserIDCollectionEnabled is currently set to FALSE so you're sending app events without collecting Advertiser ID. This can affect the quality of your advertising and analytics results.
02-07 11:13:29.409 +0000 10087 23205 23262 I DpmTcmClient: RegisterTcmMonitor from: $Proxy1
02-07 11:13:29.420 +0000 10087 23205 23262 E GraphResponse: GraphRequest can't be used when Facebook SDK isn't fully initialized
02-07 11:13:29.421 +0000 10087 23205 23265 E GraphResponse: GraphRequest can't be used when Facebook SDK isn't fully initialized
02-07 11:13:29.426 +0000  1000  1549  2311 D ConnectivityService: NetReassign [no changes]
02-07 11:13:29.436 +0000 10087 23205 23262 E GraphResponse: GraphRequest can't be used when Facebook SDK isn't fully initialized
02-07 11:13:29.459 +0000 10087 23205 23205 I FIAM.Headless: Starting InAppMessaging runtime with Installation ID fzl2U9bTRrWkm8EZDu4Nbv
02-07 11:13:29.466 +0000 10087 23205 23262 E GraphResponse: GraphRequest can't be used when Facebook SDK isn't fully initialized
02-07 11:13:29.468 +0000 10087 23205 23262 E GraphResponse: GraphRequest can't be used when Facebook SDK isn't fully initialized
02-07 11:13:29.498 +0000  1000  1549  2654 D ConnectivityService: requestNetwork for uid/pid:10087/23205 NetworkRequest [ TRACK_DEFAULT id=288, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10087 AdministratorUids: [] RequestorUid: 10087 RequestorPackageName: com.hmdglobal.support] ]
02-07 11:13:29.499 +0000  1000  1549  2311 D ConnectivityService: NetReassign [288 : null → 100]
02-07 11:13:29.500 +0000  1000  1549  2307 D UntrustedWifiNetworkFactory: got request NetworkRequest [ TRACK_DEFAULT id=288, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10087 AdministratorUids: [] RequestorUid: 10087 RequestorPackageName: com.hmdglobal.support] ] with score 60 and providerId 5
02-07 11:13:29.500 +0000  1000  1549  2307 D WifiNetworkFactory: got request NetworkRequest [ TRACK_DEFAULT id=288, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10087 AdministratorUids: [] RequestorUid: 10087 RequestorPackageName: com.hmdglobal.support] ] with score 60 and providerId 5
02-07 11:13:29.500 +0000  1000  1549  2340 D Ethernet: got request NetworkRequest [ TRACK_DEFAULT id=288, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10087 AdministratorUids: [] RequestorUid: 10087 RequestorPackageName: com.hmdglobal.support] ] with score 60 and providerId 5
02-07 11:13:29.501 +0000 radio  2747  2747 D PhoneSwitcherNetworkRequstListener: got request NetworkRequest [ TRACK_DEFAULT id=288, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED Uid: 10087 AdministratorUids: [] RequestorUid: 10087 RequestorPackageName: com.hmdglobal.support] ] with score 60 and providerId 5
02-07 11:13:29.517 +0000 10087 23205 23259 I WM-WorkerWrapper: Worker result SUCCESS for Work [ id=64dac7f1-6a47-4fc9-aabf-b03e549410e3, tags={ com.hmdglobal.support:REFRESH_TOKEN_WORKER, com.hmdglobal.support.features.login.workers.RefreshTokenWorker } ]
02-07 11:13:29.520 +0000  1000  1549  2311 D ConnectivityService: releasing NetworkRequest [ TRACK_DEFAULT id=288, [ Ca
expected not to be: 7FB4DAA9AE7782D52983BA68C755F8A8136DF593697CE8F755FF426554411365
	at com.google.android.fhir.db.impl.DatabaseEncryptionKeyProviderTest.getOrCreatePassphrase_otherAliasExists_shouldReplaceKey(DatabaseEncryptionKeyProviderTest.kt:60)
```

[encryptedDatabase_thenLostKey_shouldThrowSQLiteException](https://github.com/google/android-fhir/blob/c06c2a3c29fedd25865e63d310ffaf88befebd62/engine/src/androidTest/java/com/google/android/fhir/db/impl/EncryptedDatabaseErrorTest.kt#L102)

```
--------- beginning of main
02-07 11:29:29.386 +0000  logd     0     0 W swapper/7: type=1400 audit(0.0:5567): avc: denied { kill } for capability=5 scontext=u:r:kernel:s0 tcontext=u:r:kernel:s0 tclass=capability permissive=0
02-07 11:29:31.044 +0000 10159  2729  3732 D QCNEJ/WlanStaInfoRelay: Received action: android.net.wifi.RSSI_CHANGED
02-07 11:29:31.053 +0000 10046  3244  4401 I CarrierServices: [129] cew.d: (PEv2-SM-216099679): DisabledState: event Connectivity event [null]
02-07 11:29:31.055 +0000 radio  2747  3272 E PhoneInterfaceManager: [PhoneIntfMgr] getCarrierPrivilegeStatus: Invalid subId
02-07 11:29:31.056 +0000 10046  3244  4416 I CarrierServices: [131] cln.e: RCS Configuration storage in Bugle is disabled by p/h flag, using ProvisioningEngineStateCacheUtil.getProvisioningConfigurationForGivenSimIfExist()
02-07 11:29:31.058 +0000 10046  3244  4417 I CarrierServices: [132] bwc.c: Connected state: [1], networkType: [WIFI]
02-07 11:29:31.532 +0000  1017  1002  1002 I keystore: del USRPKEY_fhirEngineDbPassphrase 10197
02-07 11:29:31.536 +0000  1017  1002  1002 I keystore: del USRCERT_fhirEngineDbPassphrase 10197
02-07 11:29:31.539 +0000  1017  1002  1002 I keystore: del CACERT_fhirEngineDbPassphrase 10197
02-07 11:29:31.599 +0000  1017  1002  1002 I keystore: del USRPKEY_fhirEngineDbPassphrase 10197
02-07 11:29:31.601 +0000  1017  1002  1002 I keystore: del USRSKEY_fhirEngineDbPassphrase 10197
02-07 11:29:31.602 +0000  1017  1002  1002 I keystore: del USRCERT_fhirEngineDbPassphrase 10197
02-07 11:29:31.603 +0000  1017  1002  1002 I keystore: del CACERT_fhirEngineDbPassphrase 10197
02-07 11:29:31.607 +0000  1000   514   520 D DrmLibTime: got the req here! ret=0
02-07 11:29:31.607 +0000  1000   514   520 D DrmLibTime: command id, time_cmd_id = 770
02-07 11:29:31.607 +0000  1000   514   520 D DrmLibTime: time_getutcsec starts!
02-07 11:29:31.607 +0000  1000   514   520 D DrmLibTime: QSEE Time Listener: time_getutcsec
02-07 11:29:31.608 +0000  1000   514   520 D DrmLibTime: QSEE Time Listener: get_utc_seconds
02-07 11:29:31.608 +0000  1000   514   520 D DrmLibTime: QSEE Time Listener: time_get_modem_time
02-07 11:29:31.608 +0000  1000   514   520 D DrmLibTime: QSEE Time Listener: Checking if ATS_MODEM is set or not.
02-07 11:29:31.608 +0000  1000   514   520 D QC-time-services: Lib:time_genoff_operation: pargs->base = 13
02-07 11:29:31.608 +0000  1000   514   520 D QC-time-services: Lib:time_genoff_operation: pargs->operation = 2
02-07 11:29:31.608 +0000  1000   514   520 D QC-time-services: Lib:time_genoff_operation: pargs->ts_val = 0
02-07 11:29:31.608 +0000  1000   514   520 D QC-time-services: Lib:time_genoff_operation: Send to server  passed!!
02-07 11:29:31.609 +0000  1000   607   616 D QC-time-services: Daemon: Connection accepted:time_genoff
02-07 11:29:31.609 +0000  1000   607 23892 D QC-time-services: Daemon:Received base = 13, unit = 1, operation = 2,value = 0
02-07 11:29:31.609 +0000  1000   607 23892 D QC-time-services: Daemon:genoff_opr: Base = 13, val = 0, operation = 2
02-07 11:29:31.609 +0000  1000   607 23892 D QC-time-services: offset is: 0 for base: 13
02-07 11:29:31.609 +0000  1000   514   520 E QC-time-services: Receive Passed == base = 13, unit = 1, operation = 2, result = 0
02-07 11:29:31.609 +0000  1000   514   520 D DrmLibTime: QSEE Time Listener: ATS_MODEM is not set. Fallback to Android system time.
02-07 11:29:31.609 +0000  1000   514   520 D DrmLibTime: QSEE Time Listener: Retrieved Android system time: 1644233371
02-07 11:29:31.609 +0000  1000   514   520 D DrmLibTime: time_getutcsec returns 0, sec = 1644233371; nsec = 0
02-07 11:29:31.609 +0000  1000   514   520 D DrmLibTime: time_getutcsec finished! 
02-07 11:29:31.609 +0000  1000   514   520 D DrmLibTime: iotcl_continue_command finished! and return 0 
02-07 11:29:31.609 +0000  1000   514   520 D DrmLibTime: before calling ioctl to read the next time_cmd
02-07 11:29:31.613 +0000  1000   607   616 E QC-time-services: Daemon: Time-services: Waiting to acceptconnection
02-07 11:29:32.416 +0000 10197 23167 23195 E TestRunner: failed: encryptedDatabase_thenLostKey_shouldThrowSQLiteException(com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest)
02-07 11:29:32.417 +0000 10197 23167 23195 E TestRunner: ----- begin exception -----
02-07 11:29:32.464 +0000 10197 23167 23181 I droid.fhir.tes: Background young concurrent copying GC freed 395527(14MB) AllocSpace objects, 1(68KB) LOS objects, 36% free, 25MB/39MB, paused 93us total 131.707ms
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: java.lang.AssertionError: expected net.sqlcipher.database.SQLiteException to be thrown, but nothing was thrown
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.Assert.assertThrows(Assert.java:1028)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.Assert.assertThrows(Assert.java:981)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest.encryptedDatabase_thenLostKey_shouldThrowSQLiteException(EncryptedDatabaseErrorTest.kt:103)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at java.lang.reflect.Method.invoke(Native Method)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at androidx.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:154)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.Suite.runChild(Suite.java:128)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.Suite.runChild(Suite.java:27)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:388)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: 	at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2211)
02-07 11:29:32.471 +0000 10197 23167 23195 E TestRunner: ----- end exception -----
02-07 11:29:32.569 +0000 10197 23167 23195 I TestRunner: finished: encryptedDatabase_thenLostKey_shouldThrowSQLiteException(com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest)

java.lang.AssertionError: expected net.sqlcipher.database.SQLiteException to be thrown, but nothing was thrown
	at org.junit.Assert.assertThrows(Assert.java:1028)
	at org.junit.Assert.assertThrows(Assert.java:981)
	at com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest.encryptedDatabase_thenLostKey_shouldThrowSQLiteException(EncryptedDatabaseErrorTest.kt:103)
	at java.lang.reflect.Method.invoke(Native Method)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at androidx.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:154)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
	at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:388)
	at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2211)
```

[encryptedDatabase_thenLostKey_recreationStrategy_shouldRecreateDatabase](https://github.com/google/android-fhir/blob/c06c2a3c29fedd25865e63d310ffaf88befebd62/engine/src/androidTest/java/com/google/android/fhir/db/impl/EncryptedDatabaseErrorTest.kt#L153)

```
--------- beginning of main
02-07 11:29:37.114 +0000 10159  2729  3732 D QCNEJ/WlanStaInfoRelay: Received action: android.net.wifi.RSSI_CHANGED
02-07 11:29:37.122 +0000 10046  3244  4401 I CarrierServices: [129] cew.d: (PEv2-SM-216099679): DisabledState: event Connectivity event [null]
02-07 11:29:37.122 +0000 radio  2747  3272 E PhoneInterfaceManager: [PhoneIntfMgr] getCarrierPrivilegeStatus: Invalid subId
02-07 11:29:37.124 +0000 10046  3244  4416 I CarrierServices: [131] cln.e: RCS Configuration storage in Bugle is disabled by p/h flag, using ProvisioningEngineStateCacheUtil.getProvisioningConfigurationForGivenSimIfExist()
02-07 11:29:37.125 +0000 10046  3244  4417 I CarrierServices: [132] bwc.c: Connected state: [1], networkType: [WIFI]
02-07 11:29:39.396 +0000  logd     0     0 W swapper/4: type=1400 audit(0.0:5568): avc: denied { kill } for capability=5 scontext=u:r:kernel:s0 tcontext=u:r:kernel:s0 tclass=capability permissive=0
02-07 11:29:43.185 +0000 10159  2729  3732 D QCNEJ/WlanStaInfoRelay: Received action: android.net.wifi.RSSI_CHANGED
02-07 11:29:43.193 +0000 10046  3244  4401 I CarrierServices: [129] cew.d: (PEv2-SM-216099679): DisabledState: event Connectivity event [null]
02-07 11:29:43.193 +0000 10046  3244  4417 I CarrierServices: [132] bwc.c: Connected state: [1], networkType: [WIFI]
02-07 11:29:43.194 +0000 radio  2747  3272 E PhoneInterfaceManager: [PhoneIntfMgr] getCarrierPrivilegeStatus: Invalid subId
02-07 11:29:43.195 +0000 10046  3244  4416 I CarrierServices: [131] cln.e: RCS Configuration storage in Bugle is disabled by p/h flag, using ProvisioningEngineStateCacheUtil.getProvisioningConfigurationForGivenSimIfExist()
02-07 11:29:44.542 +0000  1017  1002  1002 I keystore: del USRPKEY_fhirEngineDbPassphrase 10197
02-07 11:29:44.545 +0000  1017  1002  1002 I keystore: del USRCERT_fhirEngineDbPassphrase 10197
02-07 11:29:44.547 +0000  1017  1002  1002 I keystore: del CACERT_fhirEngineDbPassphrase 10197
02-07 11:29:44.610 +0000  1017  1002  1002 I keystore: del USRPKEY_fhirEngineDbPassphrase 10197
02-07 11:29:44.612 +0000  1017  1002  1002 I keystore: del USRSKEY_fhirEngineDbPassphrase 10197
02-07 11:29:44.613 +0000  1017  1002  1002 I keystore: del USRCERT_fhirEngineDbPassphrase 10197
02-07 11:29:44.614 +0000  1017  1002  1002 I keystore: del CACERT_fhirEngineDbPassphrase 10197
02-07 11:29:44.618 +0000  1000   514   520 D DrmLibTime: got the req here! ret=0
02-07 11:29:44.618 +0000  1000   514   520 D DrmLibTime: command id, time_cmd_id = 770
02-07 11:29:44.618 +0000  1000   514   520 D DrmLibTime: time_getutcsec starts!
02-07 11:29:44.618 +0000  1000   514   520 D DrmLibTime: QSEE Time Listener: time_getutcsec
02-07 11:29:44.619 +0000  1000   514   520 D DrmLibTime: QSEE Time Listener: get_utc_seconds
02-07 11:29:44.619 +0000  1000   514   520 D DrmLibTime: QSEE Time Listener: time_get_modem_time
02-07 11:29:44.619 +0000  1000   514   520 D DrmLibTime: QSEE Time Listener: Checking if ATS_MODEM is set or not.
02-07 11:29:44.619 +0000  1000   514   520 D QC-time-services: Lib:time_genoff_operation: pargs->base = 13
02-07 11:29:44.619 +0000  1000   514   520 D QC-time-services: Lib:time_genoff_operation: pargs->operation = 2
02-07 11:29:44.619 +0000  1000   514   520 D QC-time-services: Lib:time_genoff_operation: pargs->ts_val = 0
02-07 11:29:44.619 +0000  1000   514   520 D QC-time-services: Lib:time_genoff_operation: Send to server  passed!!
02-07 11:29:44.620 +0000  1000   607   616 D QC-time-services: Daemon: Connection accepted:time_genoff
02-07 11:29:44.620 +0000  1000   607 23896 D QC-time-services: Daemon:Received base = 13, unit = 1, operation = 2,value = 0
02-07 11:29:44.620 +0000  1000   607 23896 D QC-time-services: Daemon:genoff_opr: Base = 13, val = 0, operation = 2
02-07 11:29:44.620 +0000  1000   607 23896 D QC-time-services: offset is: 0 for base: 13
02-07 11:29:44.620 +0000  1000   514   520 E QC-time-services: Receive Passed == base = 13, unit = 1, operation = 2, result = 0
02-07 11:29:44.620 +0000  1000   514   520 D DrmLibTime: QSEE Time Listener: ATS_MODEM is not set. Fallback to Android system time.
02-07 11:29:44.620 +0000  1000   514   520 D DrmLibTime: QSEE Time Listener: Retrieved Android system time: 1644233384
02-07 11:29:44.620 +0000  1000   514   520 D DrmLibTime: time_getutcsec returns 0, sec = 1644233384; nsec = 0
02-07 11:29:44.620 +0000  1000   514   520 D DrmLibTime: time_getutcsec finished! 
02-07 11:29:44.620 +0000  1000   514   520 D DrmLibTime: iotcl_continue_command finished! and return 0 
02-07 11:29:44.620 +0000  1000   514   520 D DrmLibTime: before calling ioctl to read the next time_cmd
02-07 11:29:44.621 +0000  1000   607   616 E QC-time-services: Daemon: Time-services: Waiting to acceptconnection
02-07 11:29:45.559 +0000 10197 23167 23195 E TestRunner: failed: encryptedDatabase_thenLostKey_recreationStrategy_shouldRecreateDatabase(com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest)
02-07 11:29:45.559 +0000 10197 23167 23195 E TestRunner: ----- begin exception -----
02-07 11:29:45.573 +0000 10197 23167 23195 E TestRunner: expected to be empty
02-07 11:29:45.573 +0000 10197 23167 23195 E TestRunner: but was: [org.hl7.fhir.r4.model.Patient@9ec141d]
02-07 11:29:45.573 +0000 10197 23167 23195 E TestRunner: 	at com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest$encryptedDatabase_thenLostKey_recreationStrategy_shouldRecreateDatabase$1.invokeSuspend(EncryptedDatabaseErrorTest.kt:199)
02-07 11:29:45.573 +0000 10197 23167 23195 E TestRunner: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
02-07 11:29:45.573 +0000 10197 23167 23195 E TestRunner: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
02-07 11:29:45.573 +0000 10197 23167 23195 E TestRunner: 	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
02-07 11:29:45.573 +0000 10197 23167 23195 E TestRunner: 	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
02-07 11:29:45.573 +0000 10197 23167 23195 E TestRunner: 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
02-07 11:29:45.573 +0000 10197 23167 23195 E TestRunner: 	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source:1)
02-07 11:29:45.573 +0000 10197 23167 23195 E TestRunner: 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
02-07 11:29:45.573 +0000 10197 23167 23195 E TestRunner: 	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source:1)
02-07 11:29:45.573 +0000 10197 23167 23195 E TestRunner: 	at com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest.encryptedDatabase_thenLostKey_recreationStrategy_shouldRecreateDatabase(EncryptedDatabaseErrorTest.kt:154)
02-07 11:29:45.573 +0000 10197 23167 23195 E TestRunner: ----- end exception -----
02-07 11:29:45.605 +0000 10197 23167 23195 I TestRunner: finished: encryptedDatabase_thenLostKey_recreationStrategy_shouldRecreateDatabase(com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest)

expected to be empty
but was: [org.hl7.fhir.r4.model.Patient@9ec141d]
	at com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest$encryptedDatabase_thenLostKey_recreationStrategy_shouldRecreateDatabase$1.invokeSuspend(EncryptedDatabaseErrorTest.kt:199)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source:1)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source:1)
	at com.google.android.fhir.db.impl.EncryptedDatabaseErrorTest.encryptedDatabase_thenLostKey_recreationStrategy_shouldRecreateDatabase(EncryptedDatabaseErrorTest.kt:154)

```

Phone spec:
Nokia 4.2
RAM 3GB
Model TA-1157
Android version 11
Build number 00EEA_3_240
Kernal version 4.9.227-perf+

--- a/engine/src/main/java/com/google/android/fhir/db/impl/DatabaseEncryptionKeyProvider.kt
+++ b/engine/src/main/java/com/google/android/fhir/db/impl/DatabaseEncryptionKeyProvider.kt
@@ -74,7 +74,7 @@ internal object DatabaseEncryptionKeyProvider {
             keyGenerator.generateKey()
           }
       hmac.init(signingKey)
-      val key = hmac.doFinal("".toByteArray(StandardCharsets.UTF_8))
+      val key = hmac.doFinal(MESSAGE_TO_BE_SIGNED.toByteArray(StandardCharsets.UTF_8))
       keyMap[keyName] = key
       return key
     } catch (exception: KeyStoreException) {
@@ -94,4 +94,5 @@ internal object DatabaseEncryptionKeyProvider {
    * encryption secret.
    */
   @VisibleForTesting const val ANDROID_KEYSTORE_NAME = "AndroidKeyStore"
+  private const val MESSAGE_TO_BE_SIGNED = "Android FHIR SDK rocks!"
 }

--- a/engine/src/main/java/com/google/android/fhir/db/impl/DatabaseEncryptionKeyProvider.kt
+++ b/engine/src/main/java/com/google/android/fhir/db/impl/DatabaseEncryptionKeyProvider.kt
@@ -94,5 +94,11 @@ internal object DatabaseEncryptionKeyProvider {
    * encryption secret.
    */
   @VisibleForTesting const val ANDROID_KEYSTORE_NAME = "AndroidKeyStore"
+  /**
+   * A message used to derive the database encryption key.
+   *
+   * The content of the message doesn't matter. However, once it is selected, we must not change it
+   * unless we introduce a mechanism to change the database encryption key.
+   */
   private const val MESSAGE_TO_BE_SIGNED = "Android FHIR SDK rocks!"
 }

